foca/api/register_openapi.py

```diff
@@ -88,3 +97,24 @@ def register_openapi(
         logger.info(f"API endpoints specified in '{spec.path_out}' added.")
 
     return app
+
+
+def _mergeSpecs(original, *args, **kwargs):
```

Comment:
Lacks typing info and docstring. Also, the function name does not conform to conventions (use snake case for all Python functions, i.e., `_merge_specs()`). I also don't think this function needs to be private - in fact it may be useful for people to use, so better call `merge_specs()`. As you don't seem to make use of `kwargs`, there's no need to include it in the signature. Finally, I think just `args`, without `original` is enough, then use `.pop(0)` to retrieve it and remove it from the list. This makes it cleaner and easier to use. Just document this behavior in the docstring (not in the summary line, but in the extended description before the `Args:` section.

You should also think about edge cases: what if the user supplies no or only one argument?
E.g.: if no arguments are supplied, return `None`, if one arg is supplied, return the unmodified specs (which is what happens already). Given these cases, it might actually make more sense to just call the function `parse_specs()`.In that case, you can then further simplify the logic in the calling context like this::

```py
if isinstance(spec.path, str):  # this takes care of the case where the user provides `str` rather than a list of `str`
    spec_path = [spec_path]
elif len(spec.path) > 1:
    spec_modified = True
spec_parsed = parse_specs(*spec.path)
```